gcloud/datastore/batch.py

```diff
@@ -189,8 +189,8 @@ def delete(self, key):
         if not _dataset_ids_equal(self._dataset_id, key.dataset_id):
             raise ValueError("Key must be from same dataset as batch")
 
-        key_pb = key.to_protobuf()
-        helpers._add_keys_to_request(self.mutation.delete, [key_pb])
+        key_pb = helpers._prepare_key_for_request(key.to_protobuf())
```

Comment:
`_add_keys_to_request` was only used here and in the place it is newly defined (in `connection`). `_add_keys_to_request` was added to be used with lists, so for one `key_pb` here `CopyFrom` suffices.

Removing the `datastore.helpers` in `connection` was made easier by moving the definition of `_add_keys_to_request` there.